backend/src/search/pinecone.go

```diff
@@ -5,32 +5,77 @@ import (
 	"fmt"
 	"net/http"
 
+	"github.com/garrettladley/mattress"
+	"gorm.io/gorm"
+
 	"github.com/goccy/go-json"
 	"github.com/gofiber/fiber/v2"
 
 	"github.com/GenerateNU/sac/backend/src/config"
 	"github.com/GenerateNU/sac/backend/src/errors"
+	"github.com/GenerateNU/sac/backend/src/models"
 	"github.com/GenerateNU/sac/backend/src/utilities"
+
+	stdliberrors "errors"
 )
 
 type PineconeClientInterface interface {
-	Upsert(item Searchable) *errors.Error
-	Delete(item Searchable) *errors.Error
+	Upsert(items []Searchable) *errors.Error
+	Delete(items []Searchable) *errors.Error
 	Search(item Searchable, topK int) ([]string, *errors.Error)
 }
 
 type PineconeClient struct {
 	Settings     config.PineconeSettings
+	IndexName    *mattress.Secret[string]
 	openAIClient *OpenAIClient
 }
 
+// Connects to an existing Pinecone index, using the host and keys provided in settings.
 func NewPineconeClient(openAIClient *OpenAIClient, settings config.PineconeSettings) *PineconeClient {
 	return &PineconeClient{
 		Settings:     settings,
 		openAIClient: openAIClient,
 	}
 }
 
+// Seeds the pinecone index with the clubs currently in the database.
+func (c *PineconeClient) Seed(db *gorm.DB) error {
+	var clubs []models.Club
+
+	if err := db.Find(&clubs).Error; err != nil {
+		return err
+	}
+
+	searchables := make([]Searchable, len(clubs))
+	for i := range clubs {
+		searchables = append(searchables, &clubs[i])
+	}
+
+	var chunks [][]Searchable
+	chunkSize := 50
+
+	for i := 0; i < len(searchables); i += chunkSize {
+		end := i + chunkSize
+
+		if end > len(searchables) {
+			end = len(searchables)
+		}
+
+		chunks = append(chunks, searchables[i:end])
+	}
+
+	for i, chunk := range chunks {
+		print(fmt.Sprintf("Uploading chunk #%d (of %d) to pinecone...\n", i+1, len(chunks)))
+		err := c.Upsert(chunk)
+		if err != nil {
+			return stdliberrors.New("club upsert failed")
```

Comment:
i don't like our current errors situation, but, for consistency should we create a use one of our custom errors here?